Neos.Flow/Tests/FunctionalTestCase.php

```diff
@@ -369,7 +379,14 @@ protected function registerRoute($name, $uriPattern, array $defaults, $appendExc
         if ($httpMethods !== null) {
             $route->setHttpMethods($httpMethods);
         }
-        $this->router->addRoute($route);
+
+        // the old router->addRoute prepended the routes so this is needed for consistency
+        $this->routes = $this->routes->withPrependedRoute($route);
```

Comment:
the register route will now always clear the default routes as the default provider from the testing context is not used. I imagined that we would merge the runtime registered routes with the actual.

And a possibly way to archive this is to have a `TestingRoutesProvider` with a static $routes cache ... we of course have to reset it on `shutDown`

Reply:
Not really sure what "default" routes for a functional testcase would be. To me it totally makes sense that the router for functional testcases  only knows the routes he is explicitly told about.

Reply:
if you like i could implement my suggestion

Reply:
I see use for a method like `registerDefaultRoutes` in the FuncitonalTestcase.php that is called explicitly for e2e tests. In other cases i see this as a very strange side effect.

Reply:
This is funny. Wellllll.... I am pretty sure that the behaviour as i described it is intended and wanted, but on the other hand i was surprised that the tests dont fail. I found a clue. I searched for a registered test route in `Testing/Routes.yaml` and found `neos/flow/test/http/foo` for example. This is referenced and tested in the `Neos.Flow/Tests/Functional/Mvc/RoutingTest.php`.

When running the test it actually doesnt fail but is rather skipped :D
This is caused by a check in https://github.com/neos/flow-development-collection/blob/86d95b38bd618ca68e6dd8d9c400367ccef2f232/Neos.Flow/Tests/Functional/Mvc/RoutingTest.php#L49-L61 which tries to avoid making the test fail if it is executed in a client project i suppose.

Reply:
Yes i can confirm that with my adjustments the tests are working again.